client/packages/invoices/src/Returns/DetailView/Footer/Footer.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FC, memo } from 'react';
+import {
+  Box,
+  ButtonWithIcon,
+  StatusCrumbs,
+  XCircleIcon,
+  useTranslation,
+  AppFooterPortal,
+  //   InvoiceNodeStatus,
+  useBreadcrumbs,
+  InvoiceNodeStatus,
+} from '@openmsupply-client/common';
+import { getStatusTranslator, outboundStatuses } from '../../../utils';
+import { OutboundReturnRowFragment, useReturns } from '../../api';
+
+// import { useOutbound, OutboundFragment } from '../../api';
+import { StatusChangeButton } from './StatusChangeButton';
+import { OnHoldButton } from './OnHoldButton';
+
+const createStatusLog = (invoice: OutboundReturnRowFragment) => {
+  const statusIdx = outboundStatuses.findIndex(s => invoice.status === s);
+  const statusLog: Record<InvoiceNodeStatus, null | undefined | string> = {
+    [InvoiceNodeStatus.New]: null,
+    [InvoiceNodeStatus.Allocated]: null,
+    [InvoiceNodeStatus.Picked]: null,
+    [InvoiceNodeStatus.Shipped]: null,
+    [InvoiceNodeStatus.Delivered]: null,
+    [InvoiceNodeStatus.Verified]: null,
+  };
+  if (statusIdx >= 0) {
+    statusLog[InvoiceNodeStatus.New] = invoice.createdDatetime;
+  }
+  if (statusIdx >= 1) {
+    statusLog[InvoiceNodeStatus.Allocated] = invoice.allocatedDatetime;
+  }
+  if (statusIdx >= 2) {
+    statusLog[InvoiceNodeStatus.Picked] = invoice.pickedDatetime;
+  }
+  if (statusIdx >= 3) {
+    statusLog[InvoiceNodeStatus.Shipped] = invoice.shippedDatetime;
+  }
+  if (statusIdx >= 4) {
+    statusLog[InvoiceNodeStatus.Delivered] = invoice.deliveredDatetime;
+  }
+  if (statusIdx >= 5) {
+    statusLog[InvoiceNodeStatus.Verified] = invoice.verifiedDatetime;
+  }
+  return statusLog;
+};
```

Comment:
This function is duplicated in multiple places. Should we pull out into the common invoices/utils folders?

Reply:
Yeah let's create a refactor issue for this! Might need a bit of rework to handle the different invoice types having different statuses, but agree it's a lot of duplication.

I wonder if the `statusIdx` pattern is out of date for frontend as well, like Andrei was saying yesterday for backend they're moving to matching on explicit types?

Reply:
> Yeah let's create a refactor issue for this! Might need a bit of rework to handle the different invoice types having different statuses, but agree it's a lot of duplication.

https://github.com/msupply-foundation/open-msupply/issues/3203

> I wonder if the `statusIdx` pattern is out of date for frontend as well, like Andrei was saying yesterday for backend they're moving to matching on explicit types?

It's only using the number locally as a way to determine order. I would probably do this a bit differently anyway, as part of the above refactor :)